src/SFA.DAS.FAT.Web/Views/Courses/Courses.cshtml

```diff
@@ -31,102 +32,216 @@
+                                         Model.Keyword))
+                            }
+
+                            @if (Model.ClearLinks.ContainsKey(CoursesViewModel.CoursesFilterType.Location) && Model.ClearLinks[CoursesViewModel.CoursesFilterType.Location].Any())
                             {
-                                <h4 class="das-filter__selected-sub-heading">Apprenticeship training course</h4>
-                                <a asp-querystring="@Model.ClearKeywordLink" title="Clear this filter '@Model.Keyword'" class="das-filter__tag das-breakable">&lsquo;@Model.Keyword&rsquo;</a>
+                                @await Html.PartialAsync("_RenderSelectedFilter", Tuple.Create(
+                                         "Apprentice's work location",
+                                         Model.ClearLinks[CoursesViewModel.CoursesFilterType.Location].First().Value,
+                                         Model.WorkLocationDisplayMessage))
                             }
-                            @if (Model.ClearLevelLinks.Any())
+
+                            @if (Model.ClearLinks.ContainsKey(CoursesViewModel.CoursesFilterType.Levels) && Model.ClearLinks[CoursesViewModel.CoursesFilterType.Levels].Any())
                             {
-                                <h4 class="das-filter__selected-sub-heading">Qualification level</h4>
-                                @foreach (var levelLink in Model.ClearLevelLinks)
-                                {
-                                    <a class="das-filter__tag" asp-querystring="@levelLink.Value" title="Clear this level filter @levelLink.Key">@levelLink.Key</a>
-                                }
+                                @await Html.PartialAsync("_RenderSelectedFilterList", Tuple.Create(
+                                         "Qualification level",
+                                         Model.ClearLinks[CoursesViewModel.CoursesFilterType.Levels]))
                             }
-                            @if (Model.ClearSectorLinks.Any())
+
+                            @if (Model.ClearLinks.ContainsKey(CoursesViewModel.CoursesFilterType.Categories) && Model.ClearLinks[CoursesViewModel.CoursesFilterType.Categories].Any())
                             {
-                                <h4 class="das-filter__selected-sub-heading">Apprenticeship category</h4>
-                                @foreach (var filterLink in Model.ClearSectorLinks)
-                                {
-                                    <a class="das-filter__tag" asp-querystring="@filterLink.Value" title="Clear this sector filter @filterLink.Key">@filterLink.Key</a>
-                                }
+                                @await Html.PartialAsync("_RenderSelectedFilterList", Tuple.Create(
+                                         "Apprenticeship category",
+                                         Model.ClearLinks[CoursesViewModel.CoursesFilterType.Categories]))

```

Comment:
There is logic here that generates model which feeds into partial, this cannot be unit tested easily. Can we instead move this logic in to the model where it generates one single list of `Key, Link, Value`  in the order we want and in the view we can just loop through this list and generate all the clear filter links? This will allow us to have more coverage.